### PR TITLE
executor: support 'admin check table' statement for table partition

### PR DESCRIPTION
**Pull request by tiancaiamao**
## What have you changed? (mandatory)

support 'admin check table' statement for table partition

## What is the type of the changes? (mandatory)

- Improvement (non-breaking change which is an improvement to an existing feature)

## How has this PR been tested? (mandatory)

unit test

PTAL @winkyao @coocood @zz-jason

### CODE REVIEW

File: executor/admin_test.go [hunk 1]
```diff
@@ -434,4 +436,21 @@ func (s *testSuite) TestAdminCheckTable(c *C) {
 	tk.MustExec(`create table test ( a  TIMESTAMP, primary key(a) );`)
 	tk.MustExec(`insert into test set a='2015-08-10 04:18:49';`)
 	tk.MustExec(`admin check table test;`)
+
+	// Test partitioned table.
+	tk.MustExec(`drop table if exists test`)
+	tk.MustExec(`set @@tidb_enable_table_partition = 1`)
+	tk.MustExec(`create table test (
+                          a int not null,
```
Comment (ciscoxll) [inline executor/admin_test.go]: Can you align the code？

**Comment by coocood** [comment]
@tiancaiamao CI failed

**Comment by coocood** [comment]
LGTM

File: executor/executor.go [hunk 2]
```diff
@@ -372,13 +372,37 @@ func (e *CheckTableExec) Next(ctx context.Context, chk *chunk.Chunk) error {
...
+		}
+		if err != nil {
+			log.Warnf("%v error:%v", t.Name, errors.ErrorStack(err))
+			return errors.Errorf("%v err:%v", t.Name, err)
+		}
+	}
+	return nil
+}
+
+func (e *CheckTableExec) doCheckPartitionedTable(info *model.PartitionInfo, tbl table.PartitionedTable) error {
```
Comment (shenli) [inline executor/executor.go]: Can we get the `info` from `tbl.Meta().GetPartitionInfo()`?

File: executor/executor.go [hunk 3]
```diff
@@ -372,13 +372,38 @@ func (e *CheckTableExec) Next(ctx context.Context, chk *chunk.Chunk) error {
...
 		}
-		for _, idx := range tb.Indices() {
-			txn := e.ctx.Txn()
-			err = admin.CompareIndexData(e.ctx, txn, tb, idx)
-			if err != nil {
-				log.Warnf("%v error:%v", t.Name, errors.ErrorStack(err))
-				return errors.Errorf("%v err:%v", t.Name, err)
-			}
+
+		if info := tb.Meta().GetPartitionInfo(); info != nil {
```
Comment (shenli) [inline executor/executor.go]: The `info` is useless here.

**Comment by shenli** [comment]
LGTM

**Pull request merged**